azure/datalake/store/core.py

```diff
@@ -1188,7 +1192,6 @@ def flush(self, syncFlag='METADATA', force=False):
                     leaseid=self.leaseid,
                     filesessionid=self.filesessionid)
                 self.first_write = False
-            return
 
         self.buffer.seek(0)
         data = self.buffer.read()
```

Comment:
can this be:
while len(data) > self.blocksize or force:
    if(len(data) < self.blocksize)
           force=false